tests/broadcasting/test_broadcasting.py

```diff
@@ -0,0 +1,94 @@
+"""This module contains the tests to check broadcastable array inputs.
+
+This ensures that the outputs/attributes of any functions/methods are unchanged when
+broadcastable array inputs are used in place of the full size arrays.
+
+To resolve failing tests there are a few options to use in utils.py:
+    - `skip_methods`: For irrelevant methods (e.g. 1D inputs only).
+    - `ignore_outputs`: To ignore comparing specific outputs or class attributes.
+    - `defined_method_args`: To manually define specific arguments for methods.
+    - `additional_init_methods`: For classes needing to call additional methods.
+"""
+
+import warnings
+from collections.abc import Callable
+
+import pytest
+from utils import (
+    Context,
+    compare_instances,
+    comparison_string,
+    generate_args,
+    get_method_list,
+    initialise_class,
+    is_equal,
+)
+
+from pyrealm.core.experimental import ExperimentalFeatureWarning
+
+shape_full: list[tuple[int, ...]]
+shape_full = [(3, 2, 2)]
+shapes_list = [
+    [(3, 2, 2), (1, 2, 2), (3, 1, 1), (1, 1, 1)],
+    [(1, 2, 2), (3, 2, 2)],
+    [(3, 1, 1), (1, 2, 2), (3, 2, 2)],
+    [(1, 1, 1)],
+]
+method_list = get_method_list()
```

Comment:
Global constants are typically uppercased (https://peps.python.org/pep-0008/#constants) - I think that is a useful convention for following where values come from.

```suggestion
SHAPE_FULL: list[tuple[int, ...]] = [(3, 2, 2)]
SHAPES_LIST: list[list[tuple[int, ...]]] = [
    [(3, 2, 2), (1, 2, 2), (3, 1, 1), (1, 1, 1)],
    [(1, 2, 2), (3, 2, 2)],
    [(3, 1, 1), (1, 2, 2), (3, 2, 2)],
    [(1, 1, 1)],
]
METHOD_LIST = get_method_list()
```

Reply:
Agreed. I have updated this.